lib/portage/dbapi/bintree.py

```diff
@@ -771,7 +837,10 @@ def _populate_local(self, reindex=True):
 					mycat = pkg_metadata.get("CATEGORY", "")
 					mypf = pkg_metadata.get("PF", "")
 					slot = pkg_metadata.get("SLOT", "")
-					mypkg = myfile[:-5]
+					if myfile.endswith(SUPPORTED_XPAK_EXTENSIONS):
```

Comment:
Doesn't this somewhat assume that `SUPPORTED_*_EXTENSIONS` will always all be of the same length? Maybe at least add an assert for that somewhere.

Reply:
Sure, BTW this is the bug just said.